### PR TITLE
(maint) Convert cert-whitelist test to clojure

### PR DESCRIPTION
**Pull request by mullr**

### CODE REVIEW

**Comment by senior** [comment]
:+1:  will merge once the release is done

File: project.clj [hunk 1]
```diff
@@ -76,7 +76,8 @@
                  [honeysql "0.5.2"]
                  [org.clojure/data.xml "0.0.8"]
                  [com.rpl/specter "0.5.7"]
-                 [org.clojure/core.async "0.1.346.0-17112a-alpha"]]
+                 [org.clojure/core.async "0.1.346.0-17112a-alpha"]
+                 [puppetlabs/http-client "0.4.4"]]
```
Comment (ajroetker) [inline project.clj]: Should we move over to using this library instead of `clj-http`?

Comment (mullr) [inline project.clj]: yeah, probably. the api is the same for common cases. certainly it's easy enough to do for tests.

File: test/puppetlabs/puppetdb/cli/services_test.clj [hunk 2]
```diff
@@ -133,3 +123,47 @@
...
+        cert-config {:ssl-cert "test-resources/localhost.pem"
+                     :ssl-key "test-resources/localhost.key"
+                     :ssl-ca-cert "test-resources/ca.pem"}
+        config (-> (svc-utils/create-config)
+                   (assoc :jetty (merge cert-config
+                                        {:ssl-port 0
+                                         :ssl-host "0.0.0.0"
+                                         :ssl-protocols "TLSv1,TLSv1.1"}))
+                   (assoc-in [:puppetdb :certificate-whitelist] (str whitelist-file)))]
+    (.deleteOnExit whitelist-file)
```
Comment (ajroetker) [inline test/puppetlabs/puppetdb/cli/services_test.clj]: We have a utils function for creating a file that gets deleted on jvm exit I believe.

Comment (ajroetker) [inline test/puppetlabs/puppetdb/cli/services_test.clj]: Yep https://github.com/puppetlabs/puppetdb/blob/1a226695db6ed96a704173b11ab9c61d53f340ef/test/puppetlabs/puppetdb/testutils.clj#L307-L317

File: test/puppetlabs/puppetdb/cli/services_test.clj [hunk 3]
```diff
@@ -133,3 +123,47 @@
         (testing (format "%s requests are refused" (name v)))
         (is (retirement-response? v (ping v)))))))
 
+(deftest whitelist-middleware
+  (testing "should log on reject"
+    (let [wl (fs/temp-file "whitelist-log-reject")]
+      (.deleteOnExit wl)
```
Comment (ajroetker) [inline test/puppetlabs/puppetdb/cli/services_test.clj]: Similarly to my comment below we have https://github.com/puppetlabs/puppetdb/blob/1a226695db6ed96a704173b11ab9c61d53f340ef/test/puppetlabs/puppetdb/testutils.clj#L307-L317

File: test/puppetlabs/puppetdb/cli/services_test.clj [hunk 4]
```diff
@@ -133,3 +123,47 @@
...
+      (spit wl "foobar")
+      (let [f (build-whitelist-authorizer (fs/absolute-path wl))]
```
Comment (ajroetker) [inline test/puppetlabs/puppetdb/cli/services_test.clj]: Could we call this `authorizer-fn` or something more specific?

File: test/puppetlabs/puppetdb/cli/services_test.clj (hunk 3 above)
Comment (mullr) [inline test/puppetlabs/puppetdb/cli/services_test.clj]: this test was just moved from the top of the file, but OK

Comment (ajroetker) [inline test/puppetlabs/puppetdb/cli/services_test.clj]: Oops sorry I was just looking at green

File: project.clj (hunk 1 above)
Comment (ajroetker) [inline project.clj]: Word, I'll make a ticket!

**Comment by mullr** [comment]
@ajroetker updated

**Comment by ajroetker** [comment]
@senior Am I ok to merge this in?

**Pull request merged**